### PR TITLE
api/models.py: Introduce new optional parameter error_code

### PR DESCRIPTION
**Pull request by nuclearcat**
Add parameter that might help to explain infrastructure errors, if node went to state "fail".

### CODE REVIEW

File: kernelci/api/models.py [hunk 1]
```diff
@@ -47,6 +47,12 @@ class ResultValues(str, enum.Enum):
     INCOMPLETE = 'incomplete'
 
 
+class FailValues(str, enum.Enum):
```
Comment (JenySadadia) [inline kernelci/api/models.py:50]: I assume these will contain different error codes received from the lava job run.
Using `ErrorCodes` and `Node.error_code` seem more suitable to me.

Comment (nuclearcat) [reply kernelci/api/models.py]: done

File: kernelci/api/models.py [hunk 2]
```diff
@@ -148,6 +154,9 @@ class Node(DatabaseModel):
     debug: Optional[Dict[str, Any]] = Field(
         description="Debug info fields (for development purposes)"
     )
+    failcode: Optional[FailValues] = Field(
```
[the remark this answers is not in the document]
Comment (nuclearcat) [reply kernelci/api/models.py:162]: done, but i am not sure if it can be used for other conditions

File: kernelci/api/models.py [hunk 3]
```diff
@@ -47,6 +47,18 @@ class ResultValues(str, enum.Enum):
...
 
+class ErrorCodes(str, enum.Enum):
+    """Enumeration to declare values to be used for Node.error_code.
+    error_code mostly set if infrastructure error occurs (runtime internal
+    error, scheduler error, etc.). This error code might be used not only
+    for debugging and logging, but also for taking automated decisions, for
+    example, to retry scheduling the job if runtime was temporary unavailable.
+    """
+
+    JOB_PARAMS = 'job_params'
```
Comment (JenySadadia) [inline kernelci/api/models.py:58]: To make it concise,
```suggestion
    INVALID_JOB_PARAMS = 'invalid_job_params'
```

Comment (nuclearcat) [reply kernelci/api/models.py]: done

**Comment by nuclearcat** [comment]
Tested: https://staging.kernelci.org:9000/viewer?node_id=65c343edd3c7969a82814dba

**Pull request merged**